pyanaconda/ui/tui/spokes/storage.py

```diff
@@ -513,27 +526,51 @@ def apply(self):
         # True. In the case of ks installs which may not have defined any
         # partition options, autopart was never set to True, causing some
         # issues. (rhbz#1001061)
-        if not self._do_mount_assign and self.clearPartType is not None:
+        if not self._do_mount_assign:
             self.data.autopart.autopart = True
             self.data.clearpart.type = self.clearPartType
             self.data.clearpart.initAll = True
             self.data.mount.clear_mount_data()
         else:
             self.data.autopart.autopart = False
+            self.data.clearpart.type = CLEARPART_TYPE_NONE
+            self.data.clearpart.initAll = False
+
+    def _ensure_init_storage(self):
```

Comment:
Could this be done as common function somewhere else, maybe in Blivet directly.
This sounds to me like reset storage from any state to the original state, isn't this basically `Blivet.reset()`?

Reply:
``storage_initialize()`` is a bit special beast which tries to make sure it behaves more friendly than ``reset()`` in the installation environment. And this method itself is very specific to text mode and users changing their mind. GUI uses a totally different approach.